android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java

```diff
@@ -1090,6 +1168,23 @@ public void onClickImpl(View v) {
             }
         } else if (mBraveWalletButton == v && mBraveWalletButton != null) {
             maybeShowWalletPanel();
+        } else if (mYouTubePipButton == v && mYouTubePipButton != null) {
+            // Prevent PIP action from being
+            // fired twice on same tab in case PIP icon is
+            // quickly tapped multiple times.
+            if (mYouTubePipTabId != -1) {
```

Comment:
this is not safe because the tab could change or close before entering fullscreen mode. The tab may also not enter fullscreen mode for because of some error and then it will never get set back to `-1`

Reply:
also this needs to be cleared when the tab navigates

Reply:
I've reworked this logic and introduced a new implementation based on `PageUserData`. 
Now the checks is handled by the tab helper that uses `PageUserData`. 
On the Android side there's a new JNI method `BraveYouTubeScriptInjectorNativeHelperJni.get().hasFullscreenBeenRequested(webContents)` that returns the state held by the tab helper.
The commit 👉 [`ac67726` (#29431)](https://github.com/brave/brave-core/pull/29431/commits/ac6772603914a15fe6c589d8edc1364beb7e25f4).
I've also introduced a new batch of tests 👉 [`79523bf` (#29431)](https://github.com/brave/brave-core/pull/29431/commits/79523bf0442cb5e645dfbfee6b4f46af1f35dbc6)